### PR TITLE
ComparisonMismatchSummary now returns ValidationMessages

### PR DESCRIPTION
**Pull request by dave-virden**
* Transformation is now carried out on the DTO object

### CODE REVIEW

**Comment by james-baker-digital-justice** [comment]
I would suggest a test that checks it serialises some ValidationMessage as well as empty list.

File: run-local.sh [hunk 1]
```diff
@@ -24,6 +24,9 @@ export DB_PASS=calculate-release-dates
 # export HMPPS_AUTH_URL=https://sign-in-dev.hmpps.service.justice.gov.uk/auth
 export HMPPS_AUTH_URL=http://localhost:9090/auth
 
+# export PRISON_API_URL=https://prison-api-dev.prison.service.justice.gov.uk
+export PRISON_API_URL=http://localhost:8080
+
```
Comment (james-baker-digital-justice) [inline run-local.sh:29]: This seems unrelated to this change, just checking if you meant to include it. (otherwise LGTM)

Comment (dave-virden) [reply run-local.sh:29]: Yes, I'd fallen over it a couple of times and left it in as it makes it easier to setup the local environment if you're not familiar with it.

**Review by james-baker-digital-justice — APPROVED** [review]
